### ISSUE TITLE
Search bar in ME terminal is broken

### ISSUE THREAD

**Issue by CatsLeftEar** [GTNewHorizons/GT-New-Horizons-Modpack#18285]
### Your GTNH Discord Username

Keyla3300

### Your Pack Version

stable

### Your Server

Epsilon

### Java Version

Java 8

### Type of Server

None

### Your Expectation

Search bar work propely and show what i search for

### The Reality

It shows related to the search items, but the search string is now very weirdly acting with what is being shown
![image](https://github.com/user-attachments/assets/b8a17521-4eaa-498c-9522-05ae4ef026a3)
Here i wrote "stone" three times, thats what items it shows me
![image](https://github.com/user-attachments/assets/19cbcaa6-a458-41fd-97b4-322966d070d4)
"BF BF EB BF EB" shows, well, ebf.
I suppose it thinks of a spacebar as another string and not continuation of the initial search

### Your Proposal

fix? idk if its intentional, but i doubt that

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by CatsLeftEar** [comment]
Update: its not ae2, its NEI

**Comment by slprime** [comment]
NEI in latest version splits the search string into tokens by spaces. This makes it possible to write search query queries using multiple tokens such as `-@#$&`

To prevent this, you need to wrap the string in quotes.

**Comment by boubou19** [comment]
quite annoying tbh, if i want to search for `creeper head` i should only see creeper heads not whatever matching creeper and head.

**Comment by slprime** [comment]
before it find items what have `creeper head`, now it find items what have `creeper` and `head`

**Comment by Glease** [comment]
just for a bit of clarity, is it search terms connected with OR or AND?

**Comment by chochem** [comment]
a space is now a AND

**Comment by chochem** [comment]
> quite annoying tbh, if i want to search for `creeper head` i should only see creeper heads not whatever matching creeper and head.

![image](https://github.com/user-attachments/assets/c50a2e34-db28-4081-b3ab-0a2e5363b42a)
yea there is nothing else.

**Comment by slprime** [comment]
> just for a bit of clarity, is it search terms connected with OR or AND?

before `space` use as `space` now as AND

**Comment by boubou19** [comment]
nothing wrong in this image:
![image](https://github.com/user-attachments/assets/dad48c21-4bec-4eaf-8de8-3ed5db924381)

but as soon as i add a w in the search:
![image](https://github.com/user-attachments/assets/14469d3e-1935-4deb-ab15-3a1f8b10124e)

Nothing changes because the wafer is correct and the boule despite not being correct, is still displayed, because in the tooltip it matches the w of the "raw ore"

**Comment by CatsLeftEar** [comment]
> NEI in latest version splits the search string into tokens by spaces. This makes it possible to write search query queries using multiple tokens such as `-@#$&`
> 
> To prevent this, you need to wrap the string in quotes.
> 
> https://discordapp.com/channels/181078474394566657/295669878222880769/1255915413020086348

Yeah, that needs a fix

**Comment by boubou19** [comment]
We don't need a fix, we need a button to swap between AND and "space" treatment of the space in NEI

**Comment by slprime** [comment]
I will add a setting that should change this behavior.

**Comment by CatsLeftEar** [comment]
> I will add a setting that should change this behavior.

maybe a button that would remove search by flavor text in description, because most of the multiblock controllers have "redstone" in their description, as well as most of the covers, a lot of buildcraft items like chipsets etc. Even without spacebars, all that stuff is shown when you search for both "redstone" and "stone", as well as "red"

**Comment by slprime** [comment]
![image](https://github.com/user-attachments/assets/05bc6dea-9fc4-4a4e-a2cd-161a49853fbd)
change this to `Prefix`

**Comment by boubou19** [comment]
yeah but that's a ductape fix for the meantime, as i am used to do tooltip search. I'd be more than happy to just change the space treatment in the searchbar

**Comment by chochem** [comment]
a config is fine but please dont change it. space as AND has been a huge improvement compared to the old where you needed to use wildcard and still got screwed by incorrect order.

**Comment by slprime** [comment]
I don't know English, so I wanted you to help me with the name of the settings.

I added settings with 3 states:
1. space works like <space> - like it wotks before NEI search refactoring
2. space works like AND - like now
3. space works like AND, but tokens must be in one search type (name, tooltip, etc.)

For now I've called it this:
```
nei.options.inventory.search.spaceMode=Space Works as
nei.options.inventory.search.spaceMode.0=AND
nei.options.inventory.search.spaceMode.1=Space
nei.options.inventory.search.spaceMode.2=Extended
```

![image](https://github.com/user-attachments/assets/80ef11a3-47ea-4105-a9e5-cfeca4220e0a)

### Example:
![image](https://github.com/user-attachments/assets/d8ea966e-89d2-4ff6-897a-f2ccd6000bce)

AND (search tokens in tooltip or title)
![image](https://github.com/user-attachments/assets/d8da8c98-76a4-43ec-a0da-f041f9133084)

Space (search exact match. in example found in tooltip)
![image](https://github.com/user-attachments/assets/55c138b8-8ebb-4edc-ba27-c51f8b7d3a7c)

Extended (search for tokens in one type. In the example, 2 tokens were in the tooltip)
![image](https://github.com/user-attachments/assets/c23125e6-0d32-42b5-a3ce-87f8c8fe76b3)

**Comment by chochem** [comment]
config was added
